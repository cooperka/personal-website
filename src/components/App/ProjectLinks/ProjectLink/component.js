import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles, Typography, Card, CardContent } from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  container: {
    width: 320,
    margin: theme.spacing(1),
  },
  anchor: {
    margin: 5,
    padding: 5,
    display: 'inline-block',
  },
  title: {
    fontSize: 18,
  },
}));

const ProjectLink = ({ name, description, url }) => {
  const classes = useStyles();

  return (
    <Card className={classes.container}>
      <CardContent>
        <Typography className={classes.title}>
          <a className={classes.anchor} href={url}>
            {name}
          </a>
        </Typography>
        <Typography variant="body2">{description}</Typography>
      </CardContent>
    </Card>
  );
};

ProjectLink.propTypes = {
  name: PropTypes.string,
  description: PropTypes.string,
  url: PropTypes.string,
};

export default ProjectLink;
