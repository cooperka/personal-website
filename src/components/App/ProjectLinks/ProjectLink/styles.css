.project-link-item {
  display: inline-block;
}

.project-link-anchor {
  margin: 0.5rem;
  padding: 0.5rem;
  display: inline-block;
}

.project-link-item .title {
  font-size: 1.8rem;
}

.project-link-item .subtitle {
  font-size: 1.4rem;
}
