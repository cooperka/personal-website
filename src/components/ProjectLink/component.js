// @flow

import React from 'react';

import './styles.css';

type Props = {
  name: string,
  description: string,
  url: string,
};

const ProjectLink = (props: Props) => {
  const { name, description, url } = props;

  return (
    <li className="project-link-item">
      <div className="title">
        <a className="project-link-anchor" href={url}>
          {name}
        </a>
      </div>
      <div className="subtitle">{description}</div>
    </li>
  );
};

export default ProjectLink;
