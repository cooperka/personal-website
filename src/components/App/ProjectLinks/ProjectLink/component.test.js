import React from 'react';
import renderer from 'react-test-renderer';

import ProjectLink from './component';

it('renders', () => {
  const tree = renderer.create(
    <ProjectLink name="name" description="description" url="url" />,
  );
  expect(tree.toJSON()).toMatchSnapshot();
});
