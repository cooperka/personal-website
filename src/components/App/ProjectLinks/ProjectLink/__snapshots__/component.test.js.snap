// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<div
  className="MuiPaper-root MuiCard-root makeStyles-container-1 MuiPaper-elevation1 MuiPaper-rounded"
>
  <div
    className="MuiCardContent-root"
  >
    <p
      className="MuiTypography-root makeStyles-title-3 MuiTypography-body1"
    >
      <a
        className="makeStyles-anchor-2"
        href="url"
      >
        name
      </a>
    </p>
    <p
      className="MuiTypography-root MuiTypography-body2"
    >
      description
    </p>
  </div>
</div>
`;
